python/ray/dataframe.py

```diff
@@ -0,0 +1,467 @@
+import pandas as pd
+import numpy as np
+import ray
+
+ray.register_custom_serializer(pd.DataFrame, use_pickle=True)
```

Comment:
this is to work around #1305?

Reply:
Yes, it turns out the `pd.core.indexes.base.Index` was also giving problems.